src/main/java/hudson/plugins/emailext/plugins/content/AbstractEvalContent.java

```diff
@@ -159,15 +160,14 @@ public static boolean isChildOf(final FilePath potentialChild, final FilePath pa
         return parent.act(new IsChildFileCallable(potentialChild));
     }
 
-    private InputStream getManagedFile(String fileName) throws UnsupportedEncodingException {
+    private InputStream getManagedFile(Run<?, ?> run, String fileName) throws UnsupportedEncodingException {
         InputStream stream = null;
         Plugin plugin = Jenkins.get().getPlugin("config-file-provider");
         if (plugin != null) {
             Config config = null;
-            ExtensionList<ConfigProvider> providers = ConfigProvider.all();
-            ConfigProvider provider = providers.get(getProviderClass());
-            assert provider != null;
-            for (Config c : provider.getAllConfigs()) {
+            assert getProviderClass() != null;
+            List<Config> configs = ConfigFiles.getConfigsInContext(run.getParent().getParent(), getProviderClass());
```

Comment:
Does this work for ALL cases? Why .getParent().getParent()? What if there is folder nesting, does it work in that case?

Reply:
Yes `run.getParent().getParent()` is just the way to obtain the `run`'s job and then the job's "container" (called `ItemGroup`), whether it's a nested subfolder or the top level.

By the way, that `provider.getAllConfigs()` usage is deprecated since long time, and [`ConfigFiles.getConfigsInContext(...)` usage](https://github.com/jenkinsci/config-file-provider-plugin/blob/master/src/main/java/org/jenkinsci/plugins/configfiles/ConfigFiles.java#L48) suggested as replacement [since 3 years](https://github.com/jenkinsci/config-file-provider-plugin/commit/bba752a45fa7c35824dcf516df1b270f4c658a5f#diff-c8a9d185f81b3702118c7cccc1c5c98cR133) ; the context here being the folder (the previously mentioned `ItemGroup`) from where the execution occurs.

Note that, therefore, [all existing cases already unit-tested passed](https://ci.jenkins.io/job/Plugins/job/email-ext-plugin/job/PR-196/lastCompletedBuild/testReport/hudson.plugins.emailext.plugins.content/).